arrays/arrays.c

```diff
@@ -83,14 +107,34 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
+  if (index > arr->count) {
+    switch(errno) {
+      default:
+        fprintf(stderr, "Inserting into array -> Error: Index is out of bounds \n");
+    }
+
+    exit(1);
+  }
 
   // Resize the array if the number of elements is over capacity
+  if (arr->count + 1 > arr->capacity) {
+    resize_array(arr);
+  }
 
   // Move every element after the insert index to the right one position
+  for (int i = index; i < arr->count; i++) {
```

Comment:
I found it easier to work from the end of the array (the far right side) down towards the index instead of the other way (i.e. for (int i =  arr->count; i > index; i--) and arr->elements[i] = arr->elements[i-1]. I don't want to say it's strictly necessary but I think it may allow you to both pass the test and have the correct print out. I think it may be because of the nature of the middle insert. At any rate maybe you could try it and see if  it works. Notice I never touch the index. I use the condition `i > index` as opposed to ">=". Just a suggestion but might be worth a try. When I did it by hand on paper I ran into the same copying error starting at index and working up to arr->count.

Reply:
That actually fixed my problem! I didn't think about reversing it. I'm still perplexed as to why that works and not the the other way around, but as you said, it could be because of the middle insert.